aten/src/ATen/native/native_functions.yaml

```diff
@@ -1987,6 +1987,9 @@
 - func: rot90(Tensor self, int k=1, int[] dims=[0,1]) -> Tensor
   variants: function, method
 
+- func: trapz(Tensor y, Tensor x, *, int dim=-1) -> Tensor
+- func: trapz(Tensor y, *, float dx, int dim=-1) -> Tensor
```

Comment:
The numpy signature is `numpy.trapz(y, x=None, dx=1.0, axis=-1)`, why is ours different? https://docs.scipy.org/doc/numpy/reference/generated/numpy.trapz.html

Reply:
accepting both `x` and `dx` makes no sense, since in that case `dx` is just ignored. I think it is just a bug in numpy. But I can make it the same otherwise.

Reply:
I think the only differences, other than not accepting both params, are:

1. not allowing `dx` or `dim` as positional args, which doesn't make sense if we don't allow both `x` and `dx`.
1. Naming `axis` as `dim`, which I did because `dim` is a pytorch standard (and we will allow `axis` anyway through automatic arg translation)
1. Not giving a default for `dx`. I will change this.

Other than "3." above which I will change, am I missing anything?

Reply:
I agree numpy makes no sense, but I don't know what "I can make it the same otherwise" means.

Reply:
Thanks for clarifying (1), that was the point I was confused about. Changing just (3) is fine.

There's this edge case where users might specify y, x, d, and dim by position.  Something like
`numpy.trapz(y, None, dx, dim)` should work in numpy (if the documentation is right) but wouldn't work with the pytorch schema. It's probably fine that this doesn't work, as long as our error message for it is clear